File: LeetCode/27RemoveElement/RemoveElement.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        # Partition method -> Time: O(n), Space: O(1)
        k = 0
        for i in range(len(nums)):
            if nums[i] != val:
                nums[k] = nums[i]
                k += 1
        return k

        # # Brute force -> Time: O(n^2), Space: O(1)
        # while val in nums:
        #     nums.remove(val)
        # return len(nums)

File: LeetCode/27RemoveElement/RemoveElement.cpp
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        // Partition method -> Time: O(n), Space: O(1)
        int k = 0;
        for (int i = 0; i < nums.size(); i++) {
            if (nums[i] != val) {
                nums[k] = nums[i];
                k++;
            }
        }
        return k;
    }
};